src/components/LoginScreen/index.js

```diff
@@ -1,37 +1,65 @@
-import React from "react";
+import React, { useEffect } from 'react';
+import { useTranslation } from 'react-i18next';
+import appSchema from '../../validations/app';
+import LanguageContainer from '../../components/LanguageContainer';
+import useForm from 'react-hook-form';
+import Validation from '../../components/Validation';
 import { withRouter } from "react-router";
-import { useTranslation } from "react-i18next";
+
 import styles from "./index.module.scss";
 
 const LoginScreen = props => {
   const { t } = useTranslation();
+  const { register, handleSubmit, errors, triggerValidation, setError } = useForm({
+    validationSchema: appSchema(t),
+    mode: 'onBlur',
+  });
+
+  useEffect(() => {
+    return () => errors && triggerValidation();
+  }, [t]);
 
-  const handleFormSubmit = event => {
-    event.preventDefault();
+  const onSubmit = data => {
+    setError(
+      "username",
```

Comment:
You do something wrong. This basic example doesn't contain all the manual things you do:
https://github.com/react-hook-form/react-hook-form/blob/master/examples/basicValidation.tsx